### ISSUE TITLE
Objective Evaluation

### ISSUE THREAD

**Issue by vivek-sethia** [CSTR-Edinburgh/merlin#379]
Hi,

Can anyone guide on how to perform objective evaluation using MCD??

Thanks

**Comment by seblemaguer** [comment]
Hello,

can you be more precise? Do you mean in general or you want to integrate/activate that in merlin?

**Comment by arsafi** [comment]
> Hello,
> 
> can you be more precise? Do you mean in general or you want to integrate/activate that in merlin?

Hi, I am working on a tts based on ossian frontend and merlin but i dont know how to do objective evaluation. your assistance is highly appreciated.

**Comment by RasmusD** [comment]
Hi @arsafi ,

Please do not spam all the different issues with your question, particularly on issues where your question has no relevance.

I'm replying here because this seems the most relevant thread.

However, you have replied on this thread: https://github.com/CSTR-Edinburgh/merlin/issues/165#issuecomment-447615495

In which you've copied in a previous posters answer which INCLUDES objective measures:

> I got the below objective results for duration model:
> Valid -- RMSE: 6.953 frames/phoneme; CORR: 0.746;
> Test -- RMSE: 6.585 frames/phoneme; CORR: 0.752;
>
> acoustic model:
> Valid -- MCD: 5.255 dB; BAP: 0.235 dB; F0:- RMSE: 11.295 Hz; CORR: 0.754; VUV: 6.822%
> Test -- MCD: 5.247 dB; BAP: 0.244 dB; F0:- RMSE: 12.003 Hz; CORR: 0.757; VUV: 6.111%

So, in short, Merlin already performs and reports objective measurements of the acoustic and duration model when training.

Please note, there's no guarantee that better objective measures yield better synthesised speech as synthetic speech quality is an inherently _subjective_ experience. If you wish to know more, I suggest you read the chapter on evaluation in any book on TTS and if you're after model quality objective measures I suggest reading the relevant chapter in a book on deep learning.
